Cabal-syntax/src/Distribution/PackageDescription/Configuration.hs

```diff
@@ -509,14 +511,25 @@ finalizePD
       , flagVals
       )
     where
+      appBounds :: L.HasBuildInfo a => a -> a
+      appBounds = maybe id applyDefaultBoundsToBuildInfo defBounds
+
+      appBoundsConstrs
+        :: [CondTree v [Dependency] a]
+        -> [CondTree v [Dependency] a]
+      appBoundsConstrs = case defBounds of
+        Nothing -> id
+        Just db -> map (mapTreeConstrs (applyDefaultBoundsToDependencies db))
```

Comment:
Why does `finalizePD` call `applyDefaultBoundsToBuildInfo` and `applyDefaultBoundsToDependencies` separately on the `CondTree`s, when the solver only calls `applyDefaultBoundsToCondTree`? I would have expected `Distribution.Types.DefaultBounds` to export only one function for applying default bounds and Cabal and the solver to each call it. That would help ensure that they both apply default bounds in the same way.